data for starbucks locations and function to parse through

prudencep opened the pull request:
I went through data containing information about all the starbucks locations in the world and oicked only the information that would be needed to display these locations on the map (Country name, longitude & latitude). Also created a new maps servlet to read the data from the file every time the server is requested.

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,56 @@
+  public void init() {
+    starbucksLocationsArray = new JsonArray();
+    Gson gson = new Gson();
+    Scanner scanner =
```
Let's move the scanning into a method. Usually I would prefer to abstract out large code into a more descriptive function since "init" doesn't really tell us all that is going on in it.

Maybe move the scanning out to a parseFile(String string) method?

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,56 @@
+      double lat = Double.parseDouble(cells[1]);
+      double lng = Double.parseDouble(cells[2]);
+
+      starbucksLocationsArray.add(gson.toJsonTree(new UfoSighting(state, lat, lng)));
```
Maybe consider a UfoSighting constructor that takes a String. Then in the constructor Javadoc, write the expected format of the string. This will abstract some of the more complicated code out of this method.

new UfoSighting(line);

/** 
* Some javadoc here.
* @param dataLine something descriptive
UfoSighting(String dataLine) {
  // Do the parsing here.

}

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,56 @@
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Returns Starbucks Locations as an array, e.g. [{"lat": 38.4404675, "lng": -122.7144313}]
```
The class doesn't technically only return the Starbucks locations since it also does some file parsing. It would probably be more accurate to change this comment to something along the lines of this servelet handles parsing and serving Starbucks data and move the comment about returning starbucks data down to the doGet maybe.

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,56 @@
+ */
+@WebServlet("/starbucks-data")
+public class MapsServlet extends HttpServlet {
+
```
nit: Remove the newline above JsonArray starbucksLocationArray

FluffySheep commented:
Please also fix some of your typos in your PR description

prudencep commented:
I changed the CSV file to only contain some Starbucks locations in the US and Canada for the purpose of reducing the amount of data to be parsed. Also, advised I moved the parsing code into a separate method for easier readability.

For the HTML file, The code I wrote collects the data from the /starbucks-data servlet and creates markers on the map for each given location.

![Maps2](https://user-images.githubusercontent.com/43092500/54860059-02d43d00-4ceb-11e9-8146-f9ff996ebeb7.png)

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+    new SbLocation();
+    Gson gson = new Gson();
+    String line = "";
+    Scanner scanner = 
```
Move scanning logic to a helper function? Per previous comment https://github.com/FluffySheep/CodeU2019-Team28/pull/59/files/557618e82b73a609121fcc564ed4914e9f5efbf7#r266224617

Something like 
`parseFile(String fileName)`

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+  @Override
+  public void init() {
+    starbucksLocationsArray = new JsonArray();
+    new SbLocation();
```
Why is there a newly created SbLocation that isn't used?

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+    double lng = Double.parseDouble(cells[2]);
+
+    SbLocation loc = new SbLocation(country, lat, lng);
+    return loc;
```
simplify to 
`return new SbLocation(country, lat, lng);`

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+   *Method to parse through a comma serpearted string.
+   */
+
+  public SbLocation parseFile(String line) {
```
This function is parsing a line, not the entire file. Maybe change to:

`getStarbucksLocation(String line)`

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+  JsonArray starbucksLocationsArray;
+  
+  /**
+   *Method to parse through a comma serpearted string.
```
Comments that can fit on one line should be put on one line. Comment should be descriptive of what the function is doing, parsing through a string is too vague. Consider the revision below:

`/** Method to parse through a comma separate string representing a Starbucks location. */`

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+/**
+ * Returns Starbucks Locations a JSON array, e.g. [{"lat": 38.4404675, "lng": -122.7144313}]
+ */
+
```
nit: extra line not needed.

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+
+
+/**
+ * Returns Starbucks Locations a JSON array, e.g. [{"lat": 38.4404675, "lng": -122.7144313}]
```
You didn't change the comment here. This class doesn't return a starbucks location. Certain functions do but not the entire class. Suggestion would be:

`/** A servelet that handles parsing JSON Starbucks data and serving in an HttpServletResponse. */`

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+  /**
+   *Method to parse through a comma serpearted string.
+   */
+
```
nit: extra line

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+    starbucksLocationsArray = new JsonArray();
+    new SbLocation();
+    Gson gson = new Gson();
+    String line = "";
```
`String line;` should work right? There is no need to construct an empty string here since a string is constructed below.

FluffySheep commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+    double lat;
+    double lng;
+    
+    private SbLocation() {
```
Is this constructor used? If not, then you should just remove this constructor.

linkaren commented on src/main/java/com/google/codeu/servlets/MapsServlet.java:
```diff
@@ -0,0 +1,73 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
```
nit: extra line here too?
